### PR TITLE
Add batch() method

### PR DESCRIPTION
**Pull request by RyanZim**
Simpler alternative to https://github.com/ExodusMovement/seco-keyval/pull/11 or https://github.com/ExodusMovement/seco-keyval/pull/12 & https://github.com/ExodusMovement/seco-keyval/pull/13.

Includes full test suite (let me know if I missed anything that should be included).

### CODE REVIEW

**Comment by RyanZim** [comment]
Tests will be fixed after https://github.com/ExodusMovement/seco-keyval/pull/15

**Review by mvayngrib — APPROVED** [review]

File: src/seco-keyval.test.js [hunk 1]
```diff
@@ -87,6 +87,63 @@ test('SecoKeyval open() / set() / delete() / get()', async (t) => {
...
+
+  const p1 = { name: 'JP' }
+  const p2 = { name: 'Daniel' }
+
+  await kv.batch([
+    { type: 'set', key: 'person1', value: p1 },
+    { type: 'set', key: 'person2', value: 'WRONG VALUE' },
+    { type: 'set', key: 'person2', value: p2 }, // Correct wrong value
+    { type: 'delete', key: 'person1' } // Delete key set earlier in batch
+  ])
```
Comment (kklash) [inline src/seco-keyval.test.js:133]: You don't compare the values from `kv` after they have been set here, need a test assertion to check `person1` is undefined and `person2` is equal to `p2`

File: src/seco-keyval.test.js [hunk 2]
```diff
@@ -87,6 +87,63 @@ test('SecoKeyval open() / set() / delete() / get()', async (t) => {
...
+  t.same(gp2, p2, 'person 2')
+
+  t.end()
+})
+
+test('SecoKeyval batch() operations are executed in order', async (t) => {
+  const passphrase = Buffer.from('please let me in')
+  const walletFile = tempFile()
+
+  let kv = new SecoKeyval(walletFile, { appName: 'test', appVersion: '1.0.0' })
```
Comment (kklash) [inline src/seco-keyval.test.js:122]:
```suggestion
  const kv = new SecoKeyval(walletFile, { appName: 'test', appVersion: '1.0.0' })
```

File: src/seco-keyval.test.js [hunk 3]
```diff
@@ -87,6 +87,63 @@ test('SecoKeyval open() / set() / delete() / get()', async (t) => {
...
+  const p2 = { name: 'Daniel' }
+
+  await kv.batch([
+    { type: 'set', key: 'person1', value: p1 },
+    { type: 'set', key: 'person2', value: 'WRONG VALUE' },
+    { type: 'set', key: 'person2', value: p2 }, // Correct wrong value
+    { type: 'delete', key: 'person1' } // Delete key set earlier in batch
+  ])
+
+  let kv2 = new SecoKeyval(walletFile, { appName: 'test', appVersion: '1.0.0' })
```
Comment (kklash) [inline src/seco-keyval.test.js:135]:
```suggestion
  const kv2 = new SecoKeyval(walletFile, { appName: 'test', appVersion: '1.0.0' })
```

**Comment by RyanZim** [comment]
@kklash I've marked your `let`/`const` nits as resolved, since they're copy/paste from other tests; I'll be doing a PR after this one is merged, to upgrade standard, and fix all of these at once.

**Review by kklash — APPROVED** [review]

**Comment by RyanZim** [comment]
Rebased and force-pushed to fix tests.

**Pull request merged**